test/e2e/setup.go

```diff
@@ -0,0 +1,367 @@
+		"ttl": to.StringPtr("72h"),
+	}
+
+	c.log.Infof("creating resource group %s", c.rg)
+	if _, err := c.groups.CreateOrUpdate(ctx, c.rg, azresources.Group{Location: &c.location, Tags: tags}); err != nil {
+		return err
+	}
+	c.log.Infof("waiting for successful provision of resource group %s", c.rg)
+	return wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		return c.checkResourceGroupIsReady(ctx)
+	})
+}
+
+func (c *clientSet) checkResourceGroupIsReady(ctx context.Context) (bool, error) {
+	resp, err := c.groups.CheckExistence(ctx, c.rg)
+	if err != nil {
+		return false, err
+	}
+	if resp.Response.StatusCode == http.StatusNotFound {
+		return false, nil
+	}
+	if resp.Response.StatusCode != http.StatusNoContent {
+		c.log.Debugf("resp: %v", resp.Response.StatusCode)
+	}
+	return true, nil
+}
+
+func randomAddressPrefix() (string, error) {
+	//	10.$((RANDOM & 127)).$((RANDOM & 255)).0/24
+	a, err := rand.Int(rand.Reader, big.NewInt(int64(127)))
+	if err != nil {
+		return "", err
+	}
+	b, err := rand.Int(rand.Reader, big.NewInt(int64(255)))
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("10.%s.%s.0/24", a.String(), b.String()), nil
+}
+
+func (c *clientSet) createVnet(ctx context.Context, clusterName, vnetName string) error {
+	masterPrefix, err := randomAddressPrefix()
+	if err != nil {
+		return err
+	}
+	workerPrefix, err := randomAddressPrefix()
+	if err != nil {
+		return err
+	}
+	t := &arm.Template{
+		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+		ContentVersion: "1.0.0.0",
+		Resources: []*arm.Resource{
+			{
+				Resource: &aznetwork.VirtualNetwork{
+					VirtualNetworkPropertiesFormat: &aznetwork.VirtualNetworkPropertiesFormat{
+						AddressSpace: &aznetwork.AddressSpace{
+							AddressPrefixes: &[]string{
+								"10.0.0.0/9",
+							},
+						},
+						Subnets: &[]aznetwork.Subnet{
+							{
+								SubnetPropertiesFormat: &aznetwork.SubnetPropertiesFormat{
+									AddressPrefix: to.StringPtr(masterPrefix),
+								},
+								Name: to.StringPtr(clusterName + "-master"),
+							},
+							{
+								SubnetPropertiesFormat: &aznetwork.SubnetPropertiesFormat{
+									AddressPrefix: to.StringPtr(workerPrefix),
+								},
+								Name: to.StringPtr(clusterName + "-worker"),
+							},
+						},
+					},
+					Name:     to.StringPtr(vnetName),
+					Type:     to.StringPtr("Microsoft.Network/virtualNetworks"),
+					Location: to.StringPtr(c.location),
+				},
+				APIVersion: apiVersions["network"],
+			},
+		},
+	}
+	c.log.Infof("creating vnet")
+	err = c.deployments.CreateOrUpdateAndWait(ctx, c.rg, "azuredeploy", azresources.Deployment{
+		Properties: &azresources.DeploymentProperties{
+			Template: t,
+			Mode:     azresources.Incremental,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	c.log.Debugf("assigning role to AZURE_FP_CLIENT_ID for vnet")
+	err = c.assignVnetRoleTo(ctx, os.Getenv("AZURE_FP_CLIENT_ID"))
```

Comment:
need to choose between AZURE_FP_CLIENT_ID and "f1dd0a37-89c6-4e07-bcd1-ffd3d43d8875" depending on RP_MODE - see custom.py